### DISCUSSION TITLE
QUpload @uploaded event return undefined xhr

### DISCUSSION THREAD

**Discussion by BispensGipsGebis** [quasarframework/quasar#9474] [General - Components / Directives / etc] [answered]
Hi all,

Being new to quasar I try to upload a file using qUploader and expects to be able to retrieve the response from the returned xhr

template:
```
<div class="row q-ma-md">
      <div class="col-4">
        <q-uploader
          label="Bulk upload wines"
          :url="api_path + 'upload/wine'"
          @uploaded="upload_done"
        />
      </div>
    </div>

```
It uploads fine and returns into the method 'uploaded-done' just fine. But when I try to get the response I'm told xhr is 'undefined':
```
upload_done(file, xhr) {
      let response = JSON.parse(xhr.response);
      console.log(response);
    }

```
If been trawling google, but it seems to be a first and I'm clearly doing something wrong. Any ideas will be highly appreciated.

Best, Rune

**Comment by ontwikkelfabriek** [comment, marked as answer]
@BispensGipsGebis 

The QUploader's `@uploaded` event only has `one` param. See the docs (look at the `events` tab, then `@uploaded`
https://quasar.dev/vue-components/uploader#quploader-api

```javascript
upload_done(info) {
      const xhr= info.xhr  // ( no need to do a JSON.parse since info.xhr is already an object)
      const files = info.files  // an array
      console.log("xhr object: ",info.xhr );
      console.log("files array: ",info.files );
    }

```

**Reply by BispensGipsGebis** [reply]
Thank you @ontwikkelfabriek Works like a charm. I used examples from google on an old version. DUH :)
